ote/src/clj/ote/services/users.clj

```diff
@@ -61,17 +61,19 @@
     (not (str/blank? name))))
 
 (defn save-user!
-  [db email-config requester form-data admin?]
+  [db email-config user form-data admin?]
   (if-not (valid-user-save? form-data)
     {:success? false}
     (with-transaction db
-      (let [username-taken? (and (not= (:username requester) (:username form-data))
-                              (username-exists? db {:username (:username form-data)}))
+      (let [{new-email :email
+             new-username :username} form-data
+            username-taken? (and (not= (:username user) new-username)
+                              (username-exists? db {:username new-username}))
             language (or (:language form-data) :fi)
-            email-changed? (not= (:email requester) (:email form-data))
+            email-changed? (not= (:email user) new-email)
             email-taken? (and email-changed?
-                           (email-exists? db {:email (:email form-data)}))
-            login-info (first (login/fetch-login-info db {:email (:email requester)}))
+                           (email-exists? db {:email new-email}))
+            login-info (first (login/fetch-login-info db {:email (:email user)}))
             password-incorrect? (if admin?
```

Comment:
A matter of taste, & this was already discussed, in any case making a note:
negations and inversions should be avoided in naming for better clarity.
```suggestion
            password-correct? (if admin?
                                             ( ...code...)
                                             false)
```